gemrb/core/Map.h

```diff
@@ -513,13 +515,16 @@ class GEM_EXPORT Map : public Scriptable {
 	PathNode* GetLine(const Point &start, int Steps, int Orientation, int flags);
 	PathNode* GetLine(const Point &start, const Point &dest, int speed, int Orientation, int flags);
 	/* Finds the path which leads to near d */
-	PathNode* FindPathNear(const Point &s, const Point &d, unsigned int size, unsigned int MinDistance = 0, bool sight = true);
-	/* Finds the path which leads to d */
-	PathNode* FindPath(const Point &s, const Point &d, unsigned int size, int MinDistance = 0);
+	PathNode* FindPath(const Point &s, const Point &d, unsigned int size,
+					   unsigned int minDistance = 0, bool sight = true);
+
 	/* returns false if point isn't visible on visibility/explored map */
 	bool IsVisible(const Point &s, int explored);
+
 	/* returns false if point d cannot be seen from point d due to searchmap */
-	bool IsVisibleLOS(const Point &s, const Point &d) const;
+	/* If checkWalkability is true, also returns false if there are impassable
+	 * blocks between s and d */
+	bool IsVisibleLOS(const Point &s, const Point &d, bool checkWalkability = false) const;
```

Comment:
This would be cleaner if you split the function like we discussed. I worked out my confusion.

Rename IsVisibleLOS to a new private function which will do the heavy lifting and change the new param to hold the flags (bits to check). Then create IsVisibleLOS just as a wrapper that calls it with PATH_MAP_SIDEWALL and IsWalkable that calls it with PATH_MAP_PASSABLE|PATH_MAP_TRAVEL, negating the result. This will allow us to get rid of some of the new duplicated code.